scene/3d/collision_shape.cpp

```diff
@@ -199,23 +209,32 @@ CollisionShape::~CollisionShape() {
 	//VisualServer::get_singleton()->free(indicator);
 }
 
-void CollisionShape::_create_debug_shape() {
+void CollisionShape::_update_debug_shape() {
+	if (debug_shape_dirty) {
```

Comment:
Logic should work so if you get this call it's because it's dirty, so the big if() is not needed.
Put it before in _shape_changed().